### DISCUSSION TITLE
How can I bind a hotkey to TabItem,such as press a key, and tabItem will be selected

### DISCUSSION THREAD

**Discussion by dameng324** [AvaloniaUI/Avalonia#14836] [Q&A]
I found that TabItem does not has a `HotKey` like `Button`, how can I simply implement a hotkey to TabItem?

I have tried `UserControl.KeyBindings` but It only works when my usercontrol is focused. If my tabControl is In another TabItem It will not trigger the command. It's not like the button's HotKey.

Thanks.

**Comment by timunie** [comment]
In these situations I tend to listen to KeyEvents. Needs probably tunneling in case you want to catch them while a TextBox has focus.

**Reply by dameng324** [reply]
sounds that there is no simply way to do that.

thanks for your answer.

**Reply by dameng324** [reply]
I learn the `HotKeyManager` code, and found some simply way to do that:
1. create a MyTabItem inhert `TabItem` and implement `ICommandSource` interface, then add `HotKeyManager.HotKey` to MyTabItem in xaml.
2. Make `ISelectable` supported in `HotKeyManager`. It support `ICommandSource` and `IClickableControl` for now. If avalonia team think it's make sence, I can create a PR to do that.

**Reply by dameng324** [reply]
Here is my `HotKeyTabItem` hope help someone.
```cs

public class HotKeyTabItem : TabItem , ICommandSource
{
    protected override Type StyleKeyOverride => typeof(TabItem);

    public HotKeyTabItem()
    {
        Command = new TabItemSelectCommand(this);
        CommandParameter = null;
    }

    public void CanExecuteChanged(object sender, EventArgs e)
    {
    }

    public ICommand? Command { get; }
    public object? CommandParameter { get; }

    public class TabItemSelectCommand : ICommand
    {
        private readonly TabItem _tabItem;

        public TabItemSelectCommand(TabItem tabItem)
        {
            _tabItem = tabItem;
        }

        public bool CanExecute(object? parameter)
        {
            return _tabItem.IsEffectivelyEnabled;
        }

        public void Execute(object? parameter)
        {
            _tabItem.IsSelected = true;
        }

        public event EventHandler? CanExecuteChanged;
    }
}
```
